Dockerfile

```diff
@@ -6,20 +6,26 @@ RUN \
   apt-get install -yq --no-install-recommends \
     build-essential \
     ca-certificates \
-    cmake \
     curl \
-    fuse \
     git \
     libssl-dev \
-    mercurial \
     pkg-config && \
   apt-get clean && \
   rm -rf /var/lib/apt
 RUN \
   curl -fsSL https://get.docker.com/builds/Linux/x86_64/docker-1.12.1.tgz | tar -C /bin -xz docker/docker --strip-components=1 && \
   chmod +x /bin/docker
 RUN \
-  curl -sSL https://storage.googleapis.com/golang/go1.10beta1.linux-amd64.tar.gz | tar -C /usr/local -xz && \
+  curl -sSL https://storage.googleapis.com/golang/go1.10beta1.linux-amd64.tar.gz | tar -C /tmp -xz && \
+  mkdir -p /usr/local/go && \
+  mv /tmp/go/bin /usr/local/go && \
+  mv /tmp/go/src /usr/local/go && \
+  mkdir -p /usr/local/go/pkg/tool/linux_amd64 && \
+  mv /tmp/go/pkg/include /usr/local/go/pkg && \
+  mv /tmp/go/pkg/tool/linux_amd64/compile /usr/local/go/pkg/tool/linux_amd64 && \
+  mv /tmp/go/pkg/tool/linux_amd64/asm /usr/local/go/pkg/tool/linux_amd64 && \
+  mv /tmp/go/pkg/tool/linux_amd64/link /usr/local/go/pkg/tool/linux_amd64 && \
+  rm -rf /tmp/go && \
```

Comment:
We only pull in what we need for compilation (reducing the uncompressed image size by ~500MB)

Reply:
While you're at it might as well move us to Go 1.10 proper instead of beta.

Reply:
:+1: